src/openrct2/ride/RideRatings.cpp

```diff
@@ -762,6 +767,42 @@ static void ride_ratings_calculate(Ride* ride)
         ride->ratings.nausea = max(0, ride->ratings.nausea);
     }
 #endif
+
+#ifdef ENABLE_SCRIPTING
+    auto& hookEngine = GetContext()->GetScriptEngine().GetHookEngine();
+    if (hookEngine.HasSubscriptions(HOOK_TYPE::RIDE_RATINGS_CALCULATE))
+    {
+        auto ctx = GetContext()->GetScriptEngine().GetContext();
+        auto originalExcitement = ride->excitement;
+        auto originalIntensity = ride->intensity;
+        auto originalNausea = ride->nausea;
+
+        // Create event args object
+        auto objIdx = duk_push_object(ctx);
+
+        duk_push_number(ctx, ride->id);
+        duk_put_prop_string(ctx, objIdx, "rideId");
+        duk_push_number(ctx, originalExcitement);
+        duk_put_prop_string(ctx, objIdx, "excitement");
+        duk_push_number(ctx, originalIntensity);
+        duk_put_prop_string(ctx, objIdx, "intensity");
+        duk_push_number(ctx, originalNausea);
+        duk_put_prop_string(ctx, objIdx, "nausea");
+
+        auto e = DukValue::take_from_stack(ctx);
```

Comment:
You can use `DukObject`, allows you to do this in a nicer way.

Reply:
Done